### ISSUE TITLE
Gaussian location-scale overfits the scale smooth: σ(Age) generalizes worse than constant variance on GAGurine

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#501]
## What the benchmark shows

The reference-quality test `tests/quality_vs_gamlss_gagurine_location_scale.rs` fits gam's Gaussian **location-scale** model (`mu ~ s(Age)`, `log-sigma ~ s(Age)`) on the real **GAGurine** dataset (MASS; urinary GAG vs Age, strongly heteroscedastic — spread falls sharply with age), train/test split, and scores the **held-out predictive distribution** (mean Gaussian NLL, CRPS).

Observed on the held-out set (n_train=235, n_test=79):

| model | held-out mean NLL | held-out CRPS |
|---|---|---|
| gam location-scale (μ+σ both smooth) | **3.291** | 2.469 |
| homoscedastic baseline (gam's same mean smooth + one global σ) | 3.194 | — |
| gamlss `NO()` (`mu~pb(Age), sigma~pb(Age)`) | **3.101** | 2.458 |

## Why this is a code-quality problem

The data is genuinely heteroscedastic, so modelling `σ(Age)` *should* improve held-out likelihood. Instead gam's location-scale fit is **worse out-of-sample than a constant-variance model** (3.291 > 3.194) and loses to gamlss by ~0.19 nats. That is the signature of the **scale smooth overfitting on the training set** — the `log-sigma` smooth is under-penalized / its smoothing parameter is mis-selected, so the predicted σ is miscalibrated on held-out data (CRPS is close, but NLL — which punishes miscalibrated σ in the tails — is not).

This is not a test bug: the mean smooth, the held-out split, and the σ extraction mirror the passing `quality_vs_gamlss_gaussian_location_scale.rs`; the homoscedastic and gamlss arms use the identical rows.

## Repro

```
cargo test --test quality_vs_gamlss_gagurine_location_scale -- --nocapture
```

## Likely area

`src/families/gamlss.rs` / the location-scale REML/scale-smoothing-parameter selection — the scale block's smoothing parameter appears to be selected too small (overfit) for out-of-sample calibration. Worth checking the scale-smooth penalty / λ-selection against gamlss's effective df on this dataset.

(Surfaced by the reference-quality suite; the test asserts objective held-out quality, not reproduction of gamlss's fit.)

**Comment by SauersML** [comment]
## Root-cause analysis (from code inspection)

The scale (log-σ) smooth is **under-penalized**; three contributing mechanisms:

1. **Zero-initialized smoothing-parameter seed for the scale block.** `build_two_block_exact_joint_setup` initializes the joint ρ seed to all-zeros (`src/families/gamlss.rs:~1471`, `rho0vec = Array1::<f64>::zeros(rho_dim)`), and `fit_location_scale_terms` calls it with `rho0_override = None` (`src/families/gamlss.rs:~2454`). Both blocks start completely unpenalized (λ=0), and the REML optimizer must climb from there.

2. **Asymmetric Fisher-information scaling between blocks.** Location working weight ∝ `1/σ²` (`gamlss.rs:~705`) while the log-σ working weight ∝ `2·(dlogσ/dη)²` ≈ O(2) (`gamlss.rs:~714`). On GAGurine σ ranges ~2.3–8.6, so the location block carries far higher curvature; the scale block's contribution to the REML penalty-logdet/gradient is comparatively weak.

3. **No per-block normalization of the REML smoothing-parameter gradient** (`src/solver/reml/unified.rs`, joint multi-block path). The weak scale-block gradient signal can't push λ_scale high enough, so σ(Age) overfits the training set and generalizes worse than constant variance.

**Suggested fix direction:** (a) data-driven, non-zero initial λ for the scale block (e.g. seed from its residual-variance scale), and/or (b) per-block normalization of the REML gradient by each block's Fisher-information scale, and/or (c) cross-block penalty balancing before REML — so the scale smooth is regularized as conservatively as mgcv `gaulss` / gamlss do.

(Hypothesis from reading the code paths above; worth confirming by logging the selected λ_scale / scale-block EDF on GAGurine vs gamlss's.)
